python_modules/dagster/dagster/_core/definitions/metadata/__init__.py

```diff
@@ -1289,7 +1291,26 @@ class MyMetadataEntries(NamedspacedMetadataEntries):
             if len(splits) == 2:
                 namespace, key = splits
                 if namespace == cls.namespace() and key in cls.__fields__:
-                    kwargs[key] = value.value if isinstance(value, MetadataValue) else value
+                    if isinstance(value, MetadataValue):
```

Comment:
Would prefer that we invert the conditional logic here and use `continue`. This way, we can avoid the four levels of nested conditionals.